bundles/mapping/mapmodule/plugin/mylocation/MyLocationPlugin.js

```diff
@@ -198,16 +200,113 @@ Oskari.clazz.define(
             var mobileDefs = this.getMobileDefs();
             this.removeToolbarButtons(mobileDefs.buttons, mobileDefs.buttonGroup);
         },
+        /**
+         * Check at if device is same than configured, or if not moblie is not configured then always return true.
+         * @method @private _isDevice
+         * @return {Boolean} is device same than configured
+         */
+        _isDevice: function () {
+            var conf = this.getConfig();
+            if (conf.mobileOnly === true && !Oskari.util.isMobile(true)) {
+                return false;
+            }
+            return true;
+        },
+
+        /**
+         * Toggle mode continues/single or if not plugin is not configured continues mode then use only single mode.
+         * @method @private _toggleMode
+         */
+        _toggleMode: function () {
+            var conf = this.getConfig();
+
+            // If plugin configured to use continuous mode and current mode is "continuous" then toggle mode to single
+            // (not draw user location and accurary circle and center map to user location)
+            if (this._isDevice() && conf.mode === 'continuous' && this._currentMode === 'continuous') {
```

Comment:
I would like to see this working in a way that _isDevice() is an implementation of isEnabled() (https://github.com/oskariorg/oskari-frontend/blob/9599a610e7bfb10ed7eb539e762b6efc8d514d8c/bundles/mapping/mapmodule/plugin/BasicMapModulePlugin.js#L288) and if the plugin is not enabled (mobileOnly but user has desktop) -> everything is just skipped.

Reply:
Ok, and if not enabled, also UI is not drawed ?

Reply:
It would seem logical to not show the ui if the functionality has been deactivated for desktop.